tests/metrics_test.py

```diff
@@ -152,8 +153,45 @@ class ClientMock:
         def __init__(self, hosts):
             self.transport = OsClientTests.TransportMock(hosts)
 
+    def test_config_opts_parsing_with_none(self):
+        # Tries parsing but fails because it does not detect password from ENV or Config
+        self.config_opts_parsing(None)
+
+    @mock.patch("osbenchmark.client.OsClientFactory")
+    def test_config_opts_parsing_with_config(self, client_OsClientfactory):
+        config_opts = self.config_opts_parsing("config")
+
+        expected_client_options = {
+            "use_ssl": True,
+            "timeout": 120,
+            "basic_auth_user": config_opts["_datastore_user"],
+            "basic_auth_password": config_opts["_datastore_password"],
+            "verify_certs": config_opts["_datastore_verify_certs"]
+        }
+
+        client_OsClientfactory.assert_called_with(
+            hosts=[{"host": config_opts["_datastore_host"], "port": config_opts["_datastore_port"]}],
+            client_options=expected_client_options
+        )
+
     @mock.patch("osbenchmark.client.OsClientFactory")
-    def test_config_opts_parsing(self, client_OsClientfactory):
+    def test_config_opts_parsing_with_env(self, client_OsClientfactory):
```

Comment:
A comment on what the test does will be helpful (although probably not all tests have such comments yet.)